helix-rest/src/main/java/org/apache/helix/rest/server/service/InstanceServiceImpl.java

```diff
@@ -220,6 +222,12 @@ private StoppableCheck performCustomInstanceCheck(String clusterId, String insta
       return new StoppableCheck(
           _customRestClient.getInstanceStoppableCheck(baseUrl, customPayLoads),
           StoppableCheck.Category.CUSTOM_INSTANCE_CHECK);
+    } catch (HttpHostConnectException ex) {
+      String msg = "Custom client side instance level health check for {}/{} failed "
+          + "because of connection refused.";
+      LOG.error(msg, clusterId, instanceName, ex);
+      return new StoppableCheck(false, Arrays.asList(instanceName),
+          StoppableCheck.Category.CUSTOM_INSTANCE_CHECK);
     } catch (IOException e) {
```

Comment:
Can we make all the exceptions to make the customize check fail instead of throwing exception?

Reply:
I also have the same question. But my thought was this PR is to fix null value for connection refused, so I didn't change throwing IOException. If we would like to do it, maybe a separate PR would be better?

Reply:
I remember you said connection exception is part of IOException?

Reply:
Yes.

<img width="476" alt="image" src="https://user-images.githubusercontent.com/5187721/66232474-c02ded00-e69d-11e9-910c-6082b4f24baf.png">

Did you mean doing below, instead of throwing IOException, we do returning a failed StoppableCheck?
```
    } catch (IOException ex) {
       LOG.error("Custom client side instance level health check for {}/{} failed.",
           clusterId, instanceName, ex);
       return new StoppableCheck(false, Arrays.asList(instanceName),
           StoppableCheck.Category.CUSTOM_INSTANCE_CHECK);
    }
```

Reply:
If that's the case, we dont need a separate rb but just fix the handling the exception correctly.

Reply:
That was my initial idea to just handle IOException and return StoppableCheck, instead of throwing the IOException. Already pushed a commit to address this. Can you review?